code/controllers/subsystem/spatial_gridmap.dm

```diff
@@ -0,0 +1,567 @@
+///the subsystem creates this many [/mob/oranges_ear] mob instances during init. allocations that require more than this create more.
+#define NUMBER_OF_PREGENERATED_ORANGES_EARS 2500
+
+// macros meant specifically to add/remove movables from the hearing_contents and client_contents lists of
+// /datum/spatial_grid_cell, when empty they become references to a single list in SSspatial_grid and when filled they become their own list
+// this is to save memory without making them lazylists as that slows down iteration through them
```

Comment:
assertion here is insertion/modification is less of a hot path then reading, right?

Reply:
most likely. i dont actually think this system is needed at all anymore since all our calculations for how much memory this system was taking were made when the cells were 5x5 but now theyre 17x17. meaning instead of 2601 cells per z level theres only 225

Reply:
no wait i misread this, most of the time since players move a lot im pretty sure contents move between cells more often than cells are read